synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -125,28 +109,43 @@ def __init__(self, hs: "HomeServer"):
         # Whether to support MSC3772 is supported.
         self._relations_match_enabled = self.hs.config.experimental.msc3772_enabled
 
+    async def get_rules(
+        self, room_id: str, event: EventBase
+    ) -> Dict[str, List[Dict[str, dict]]]:
+        """Given an event return the rules for all users who are
+        currently in the room.
+        """
+
+        local_users = await self.store.get_local_users_in_room(room_id)
+
+        if event.type == EventTypes.Member and event.membership == Membership.JOIN:
+            if self.hs.is_mine_id(event.state_key):
+                local_users = list(local_users)
+                local_users.append(event.state_key)
```

Comment:
Signpost comment would be helpful.

If we `event.sender` is already in the room and sends another JOIN (e.g. name change), does this mean they get double-counted in `local_users`?

Reply:
I think the database should deduplicate though.